coding-challenges.js

```diff
@@ -2,59 +2,96 @@
 
 // --------------------1) Create a function that takes in an array. Each time the function is run, remove the first item from the array and shuffle the remaining content. If the array is empty, return "The array is empty."
 
-
-
-
-
 var collections = ["purple", "blue", "green", "yellow", "pink"]
 // Expected output example1 (can be a different order): ["yellow", "blue", "pink", "green"]
 // Expected output example2 (can be a different order): ["blue", "green", "pink"]
 // Expected output example3 (can be a different order): ["pink", "green"]
 
-
-// --------------------2) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
-
+function randomRemoval(arr) {
+        if (arr.length > 1){
+        array = []
+        arr.shift()
+        while (arr.length > 0) {
+            r = Math.floor(Math.random()*arr.length)
+            array.push(arr[r])
+            arr.splice( r, 1)
+        }
+        return array
+    } else {
+        return "The array is empty"
+    }
+}
+console.log(collections = randomRemoval(collections));
+console.log(collections = randomRemoval(collections));
+console.log(collections = randomRemoval(collections));
+console.log(collections = randomRemoval(collections));
+console.log(collections = randomRemoval(collections));
 
 
 
+// --------------------2) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
 
 var cubeAndSum1 = [2, 3, 4]
 // Expected output: 99
 var cubeAndSum2 = [0, 5, 10]
 // Expected output: 1125
 
+// HECK YEAH I got it into one line
+function cubeSum (arr) { return arr.map(val => Math.pow(val, 3)).reduce((a,b) => a+b) }
 
-
-// --------------------3) Create a function that takes an array of numbers and returns an array of the minimum and maximum numbers in that order.
-
-
+console.log(cubeSum(cubeAndSum1))
+console.log(cubeSum(cubeAndSum2))
 
 
+// --------------------3) Create a function that takes an array of numbers and returns an array of the minimum and maximum numbers in that order.
 
 var nums1 = [3, 56, 90, -8, 0, 23, 6]
 // Expected output: [-8, 90]
 var nums2 = [109, 5, 9, -59, 8, 24]
 // Expected output: [-59, 109]
 
+//I DID IT AGAIN!
+function minMax (arr) { return [Math.min(...arr), Math.max(...arr)]}
```

Comment:
woohoo! just be prepared for an interviewer to ask you to write out the long-form :)